cmd/src/actions_exec.go

```diff
@@ -586,38 +590,116 @@ fragment repositoryFields on Repository {
 
 		// Skip repos from unsupported code hosts but don't report them explicitly.
 		if !includeUnsupported && strings.ToLower(repo.ExternalRepository.ServiceType) != "github" && strings.ToLower(repo.ExternalRepository.ServiceType) != "bitbucketserver" {
-			unsupported = append(unsupported, repo.Name)
+			unsupported[repo.Name] = struct{}{}
 			continue
 		}
 
-		if repo.DefaultBranch == nil || repo.DefaultBranch.Name == "" {
-			skipped = append(skipped, repo.Name)
-			continue
+		if _, ok := reposByID[repo.ID]; !ok {
+			reposByID[repo.ID] = &ActionRepo{
+				ID:   repo.ID,
+				Name: repo.Name,
+			}
 		}
+	}
 
-		if repo.DefaultBranch.Target.OID == "" {
-			skipped = append(skipped, repo.Name)
-			continue
+	reposQueryTemplate := `{
+  repositories(names: [%s]) {
+    nodes {
```

Comment:
Since this is a connection — does this always return all default branches, or do we need to check for `pageInfo.HasNextPage`?